subsys/net/ip/net_if.c

```diff
@@ -4932,14 +4932,58 @@ void net_if_call_link_cb(struct net_if *iface, struct net_linkaddr *lladdr,
 	k_mutex_unlock(&lock);
 }
 
-static bool need_calc_checksum(struct net_if *iface, enum ethernet_hw_caps caps)
+static bool need_calc_checksum(struct net_if *iface, enum ethernet_hw_caps caps, enum net_if_checksum_protocol_type chksum_type)
 {
 #if defined(CONFIG_NET_L2_ETHERNET)
+	bool have_checksum_support;
+	bool have_checksum_support_protocol;
+	struct ethernet_config config;
+	int err;
+
+	have_checksum_support_protocol = false;
+
 	if (net_if_l2(iface) != &NET_L2_GET_NAME(ETHERNET)) {
 		return true;
 	}
 
-	return !(net_eth_get_hw_capabilities(iface) & caps);
+	have_checksum_support = net_eth_get_hw_capabilities(iface) & caps;
+	err = net_eth_get_hw_config(iface,caps == ETHERNET_HW_RX_CHKSUM_OFFLOAD ? ETHERNET_CONFIG_TYPE_RX_CHECKSUM_SUPPORT : ETHERNET_CONFIG_TYPE_TX_CHECKSUM_SUPPORT, &config);
```

Comment:
`caps == ETHERNET_HW_RX_CHKSUM_OFFLOAD ? ETHERNET_CONFIG_TYPE_RX_CHECKSUM_SUPPORT : ETHERNET_CONFIG_TYPE_TX_CHECKSUM_SUPPORT` - I'd add a helper variable for this, it reads hard when inside a function call.

Reply:
OK, done.